### PR TITLE
V4.1.2 rc 1

### PR DESCRIPTION
**Pull request by red010b37**

### CODE REVIEW

**Comment by red010b37** [comment]
Please do not merge till we are 100% ready

File: src/chainparams.cpp [hunk 1]
```diff
@@ -137,6 +137,16 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_CSV_LEGACY].nStartTime = 1462060800; // May 1st, 2016
         consensus.vDeployments[Consensus::DEPLOYMENT_CSV_LEGACY].nTimeout = 1525132800; // May 1st, 2018
 
+        // Deployment of Community Fund Accumulation
+        consensus.vDeployments[Consensus::DEPLOYMENT_COMMUNITYFUND_ACCUMULATION].bit = 7;
+        consensus.vDeployments[Consensus::DEPLOYMENT_COMMUNITYFUND_ACCUMULATION].nStartTime = 1525132800; // Apr 1st, 2018
```
Comment (red010b37) [inline src/chainparams.cpp]: 1525132800 is may 1st 2018 not apr? Is this correct

File: src/chainparams.cpp [hunk 2]
```diff
@@ -137,6 +137,16 @@ class CMainParams : public CChainParams {
...
+        consensus.vDeployments[Consensus::DEPLOYMENT_COMMUNITYFUND_ACCUMULATION].nTimeout = 1554076800; // Apr 1st, 2019
+
+        // Deployment of NTP Sync
+        consensus.vDeployments[Consensus::DEPLOYMENT_NTPSYNC].bit = 8;
+        consensus.vDeployments[Consensus::DEPLOYMENT_NTPSYNC].nStartTime = 1525132800; // Apr 1st, 2018
```
Comment (red010b37) [inline src/chainparams.cpp]: 1525132800 is may 1st 2018 not 1st Apr? Is this correct

File: src/consensus/cfund.cpp [hunk 3]
```diff
@@ -37,61 +41,113 @@ void CFund::SetScriptForPaymentRequestVote(CScript &script, uint256 prequesthash
 bool CFund::FindProposal(string propstr, CFund::CProposal &proposal)
 {
 
-    return pblocktree->ReadProposalIndex(uint256S("0x"+propstr), proposal);
+    return CFund::FindProposal(uint256S("0x"+propstr), proposal);
```
Comment (red010b37) [inline src/consensus/cfund.cpp]: Just checking - There is no way this can activate until the next release

File: src/main.h [hunk 4]
```diff
@@ -96,7 +97,7 @@ static const unsigned int UNDOFILE_CHUNK_SIZE = 0x100000; // 1 MiB
 /** Maximum number of script-checking threads allowed */
 static const int MAX_SCRIPTCHECK_THREADS = 16;
 /** -par default (number of script-checking threads, 0 = auto) */
-static const int DEFAULT_SCRIPTCHECK_THREADS = 0;
+static const int DEFAULT_SCRIPTCHECK_THREADS = 2;
```
Comment (red010b37) [inline src/main.h]: Why did we limit the threads here to 2 instead of auto?

File: src/miner.cpp [hunk 5]
```diff
@@ -744,9 +750,31 @@ void NavCoinStaker(const CChainParams& chainparams)
                 MilliSleep(1000);
             }
 
+            if (nLastTime != 0 && nLastSteadyTime != 0)
```
Comment (red010b37) [inline src/miner.cpp]: Should this be split out to a function? - just a question not saying we should

File: src/chainparams.cpp (hunk 1 above)
Comment (aguycalled) [reply src/chainparams.cpp]: fixed 8f96226c59a02540c7e6dcbcecb35ad584514b80

File: src/chainparams.cpp (hunk 2 above)
Comment (aguycalled) [reply src/chainparams.cpp]: fixed 8f96226c59a02540c7e6dcbcecb35ad584514b80

File: src/consensus/cfund.cpp (hunk 3 above)
Comment (aguycalled) [reply src/consensus/cfund.cpp]: Activation of the consensus changes related to CFund is triggered by a different soft fork.

File: src/main.h (hunk 4 above)
Comment (aguycalled) [reply src/main.h]: It's default value, not max. Improves performance

File: src/miner.cpp (hunk 5 above)
Comment (aguycalled) [reply src/miner.cpp]: i don't reject the idea

**Comment by aguycalled** [comment]
closed. new pr at https://github.com/NAVCoin/navcoin-core/pull/209/

**Pull request closed without merging**